battle.go

```diff
@@ -228,3 +238,14 @@ type FightTurn struct {
 func (turn FightTurn) Priority() int {
 	return 0
 }
+
+// An item turn has the a higher priority than any move.
+type ItemTurn struct {
+	item   *Item    // Which item is being consumed
+	target *Pokemon // The target Pokemon for the item
```

Comment:
Why? We can validate that the Pokemon is part of active Pokemon when consuming the item. Keeping track of indices vs. direct references to Pokemon is better for UX imo?

Reply:
Because the agent _should_ receive a copy of the battle data, so the agent can't possibly know the pointer to the real pokemon. If it _can_ do that, then it's a bug.